Lib/test/test_super.py

```diff
@@ -422,6 +422,45 @@ def test(name):
         test("foo1")
         test("foo2")
 
+    def test_reassigned_new(self):
+        class A:
+            def __new__(cls):
+                pass
+
+            def __init_subclass__(cls):
+                if "__new__" not in cls.__dict__:
+                    cls.__new__ = cls.__new__
+
+        class B(A):
+            pass
+
+        class C(B):
+            def __new__(cls):
+                return super().__new__(cls)
+
+        C()
```

Comment:
This is generally tightly coupled with the specialization warmup being 2 right? I think we should instead use a helper function to warm it up so that the test doesnt become ineffective if the specialization boundarybchanges. IIRC, there is a `test.support` specialize warm up function or something that loops the code a set number of times.

Same for below.

Reply:
@Fidget-Spinner I can't find any such utility in `test.support`. It looks to me like the convention so far is to just have a local constant `ADAPTIVE_WARMUP_DELAY = 2` and use that to iterate. E.g. see prior art in `test_dis.py`, `test_call.py`.

For now I've updated these tests to use that same pattern.

It makes sense to me that we'd have a more centralized utility for this, but it seems better to do that in a separate PR dedicated to that purpose.

Reply:
cc @brandtbucher

Reply:
> @Fidget-Spinner I can't find any such utility in `test.support`. It looks to me like the convention so far is to just have a local constant `ADAPTIVE_WARMUP_DELAY = 2` and use that to iterate. E.g. see prior art in `test_dis.py`, `test_call.py`.
> 
> For now I've updated these tests to use that same pattern.
> 
> It makes sense to me that we'd have a more centralized utility for this, but it seems better to do that in a separate PR dedicated to that purpose.

I think I was planning to add it then I forgot about it. My bad!